api/envoy/type/matcher/string.proto

```diff
@@ -45,6 +45,13 @@ message StringMatcher {
     // * The regex *\d{3}* does not match the value *1234*
     // * The regex *\d{3}* does not match the value *123.456*
     string regex = 4 [(validate.rules).string.max_bytes = 1024];
+
+    // The input string must match if not the one exactly specified here.
+    //
+    // Examples:
+    //
+    // * *abc* only matches if the value is not *abc*.
+    string not_exact = 5;
```

Comment:
Instead of adding `not_exact`, can you just add `not`? I think we do this somewhere else (maybe RBAC) and it flows pretty nicely. Then you can negate any of the things above. I would put this outside of the oneof and make it a general negation.